src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -168,6 +173,19 @@ public void visitToken(DetailAST ast) {
                 }
                 break;
 
+            case TokenTypes.LITERAL_NEW:
+                if (ast.getFirstChild() != null) {
+                    final ClassDesc currClass = classes.peek();
+                    final String qualifiedAnonInnerClassName =
+                            packageName + PACKAGE_SEPARATOR + ast.getFirstChild().getText();
```

Comment:
Unless the code is otherwise, this does not look correct.

The qualified path of the anonymous class should include the class and any further anonymous classes inside it.
So package "a" and class "B" with anonymous class "C" should have a qualified path of "a.B.C". And if there is a anonymous class inside that of "D" it should have a qualified path of "a.B.C.D".

Reply:
Yes, totally agreed, idk how I missed that, fixed it now. Yeah, the binary name starts to become `<outerclass>$<innerclass>`.